### PR TITLE
docs: add notes on verification and proof properties

### LINKED ISSUE #54: Register JwtProof2020 or warn users not to use this library
Register it here:  

- https://w3c-ccg.github.io/ld-cryptosuite-registry/
- https://github.com/w3c-ccg/security-vocab

its 2021, where is the full IRI for JwtProof2020 ?

`https://w3id.org/did/v1` does not define `JwtProof2020`... 

`https://www.w3.org/ns/did/v1` does not define ANY proof formats (only verification methods)....

You should review https://w3c.github.io/vc-data-model/#dfn-context

and define a context which you control, where you can define all your terms for your users... see:

https://w3c.github.io/did-spec-registries/#schnorrsecp256k1verificationkey2019
https://w3c.github.io/did-spec-registries/#gpgverificationkey2020

Feel free to register your extensions in did spec registries as well.

### PR DESCRIPTION
**Pull request by mirceanis**
I added some clarification about the result of credential and presentation verification.

fixes #54

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,6 +136,7 @@ console.log(verifiedVC)
       },
     },
     "proof": {
+      // synthetic proof type, NOT a registered LD-Proof type
```
Comment (OR13) [inline README.md:139]:
```suggestion
      //  proof type for internal use, NOT a registered vc-data-model type
```

File: README.md [hunk 2]
```diff
@@ -214,10 +205,26 @@ console.log(verifiedVP)
     '@context': [Array],
     issuanceDate: '2019-09-09T16:07:43.000Z',
     proof: {
+      // synthetic proof type, NOT an LD-Proof
```
Comment (OR13) [inline README.md:208]:
```suggestion
      // proof type for internal use, NOT a registered W3C vc-data-model proof type
```

File: README.md [hunk 3]
```diff
@@ -214,10 +205,26 @@ console.log(verifiedVP)
...
+
+#### Notes on verification and proof properties
+
+The result of the verification methods, when successful, also conveniently contain the decoded and parsed payloads, in
+a format that closely matches the [W3C data model]() for verifiable credentials and presentations. 
+This makes it easier to work with both credential encodings in the same system.
+This parsed payload also shows a `proof` property that lists the full JWT credential or presentation.
+
+The `JwtProof2020` is a synthetic proof type, usable for differentiating credentials by type.
+It is not a registered LD-Proof algorithm and should not be treated as such.
```
Comment (OR13) [inline README.md:225]:
```suggestion
It is not a registered W3C VC Data Model algorithm and should not be treated as such.
```

File: README.md [hunk 4]
```diff
@@ -214,10 +205,26 @@ console.log(verifiedVP)
...
     }
   }
 }
 */
 ```
+
+#### Notes on verification and proof properties
+
+The result of the verification methods, when successful, also conveniently contain the decoded and parsed payloads, in
+a format that closely matches the [W3C data model]() for verifiable credentials and presentations.
```
Comment (OR13) [inline README.md:220]:
```suggestion
a format that closely matches the [W3C data model](https://www.w3.org/TR/vc-data-model/) for verifiable credentials and presentations. 
```

Comment (mirceanis) [reply README.md:220]: good catch, thanks :)

**Review by OR13 — COMMENTED** [review]
the main issue is the proof type "JwtProof2020" is not described here: https://www.w3.org/TR/vc-data-model/

This is the W3C Verifiable Credentials Specification... 

There are a number of LD-Proof types which are not described in it, including jws, pgp and bbs+.... 

Its more accurate to say that "JwtProof2020" is not registered in the vc data model, AND its not a registered LD-Proof type... I think folks would care less about the later than the former.

When people use the library, they want it to produce JSON that matches the W3C Spec... they probably care less about LD proofs than W3C conformance.

**Comment by mirceanis** [comment]
I agree.
I hope that these changes shed light on the fact that the JSON produced by the verification methods is only for convenience.

Verification logic is actually something like this:

```
try {
  await verifyCredential(...)
} catch (e) {
  console.error('Verification failed for reason:', e)
}
```

**Review by OR13 — APPROVED** [review]

**Pull request merged**